doc/wg/core/notes/core-notes-2021-10-29.md

```diff
@@ -0,0 +1,124 @@
+ * Phil: Right, but what if an app `loop {}` that? Never gives back buffer
+ * Amit: Right, so might need some time-based policy, etc
+ * Phil: Right, but then there's no way to gaurentee that a process won't crash
+ * Jett/Phil/Amit: There are some limits to the execution model; nothing like a signal, only push upcalls in response to yield, so limited opporunity to notify
+ * Phil: One challenge here is what revocation will look like in Rust
+ * Jett: At a certain point, if an app is seeming malicious (or erroneous), just shut down the app
+ * Leon: That is a very dangerous interface
+ * <yeah>
+ * Jett: The goal is to make sure an app can't starve / lock a shared resource; eventually have to kill them
+ * Amit: If only we had virtual memory
+ * <yeah>
+ * Leon: Is memory that's re-allowed into the kernel, is that a kernel resource in that timeframe? Could it then re-allow this buffer into yet another application?
+ * Jett: Kernel to app is now app-ish memory, so when app allows to kernel, it looks like regular app-allowed memory
+ * Leon: And the kernel can't allow the original buffer to another app?
+ * Jett: Right, kernel has given up the buffer
+ * Phil: Whatever this mechanism is, it should look like IPC; it's shared memory with the kernel
+ * Phil: It's likely a success if IPC is this same mechansim
+ * Jett: What happens today with IPC out curiousity? If A shares to B, can B share A's region to C?
+ * Amit: Belive IPC checks that a share is from app-owned memory
+ * Jett: Makes sense
+ * Amit: And just that friendly old reminder, that IPC is what it is to get a paper out the door, not because it was actually the right design
+ * Leon: One challenge is that most of the UKB is async; so the stack frame mechanism would likely be hard to implement in practice; it might be that use cases for this are very narrow
+ * Jett: Agree, that most interfaces are async, but they are logically sync
+ * Leon: That's right, but if I get a, say, 500 byte buffer, but flash only accepts 100 byte writes, the interface requires that I yield and go to different stack frame; so monitoring the stack will not work
+ * Jett: So, taking your example with 100 byte chunks, this could all work off the same nested dispatch upcall, no?
+ * Loen: Oh, yeah, but does that get really hard for the kernel to enforce
+ * Jett: Don't think so, once the dispatch stack frame is popped, the buffer is returned to the kernel
+ * Phil: Think the takeway is that so long as kernel has final control, there's nothing philosophically opposed, but it's going to be very subtle / tricky to implement
+ * Jett: {missed}
+ * Phil: If the shared buffer is associated with a particular device ID, and only available to that device ID, then it gets easier
+ * Phil: What gets tricky is if it is an unlabeled buffer, and moved around
+ * Jett: I think if allow ro/rw moves into the kernel core, then this simplifies
+ * Leon: Don't have to do exhaustive searches if it's time-based
```

Comment:
```suggestion
 * Leon: Don't have to do exhaustive searches if it's time-based, because if we decide to forcefully kill the process in case it violates these constraints, we can make use of the regular process cleanup mechanisms (e.g. `ProcessBuffer`s won't be accessible any longer)
```